### PR TITLE
Update Feature

### PR DESCRIPTION
**Pull request by James-A1**
- New get primary key function on schema
- Update function in mapper
- Update function test

### CODE REVIEW

File: tests/CoreTest.php [hunk 1]
```diff
@@ -110,4 +110,44 @@ public function insertShouldWork()
...
+		unset($updatedEntityArray["id"]);
+
+		$entity = CoreEntity::fromRecord(
+			CoreEntity::fromRecord(CoreEntity::createInstance(), $originalEntity->toRecord($originalEntity), []),
+			$updatedEntityArray
+		);
+		$updatedEntity = $mapper->update($entity);
+
+		$this->assertEquals($originalEntity->id, $updatedEntity->id);
+		$this->assertNotEquals($originalEntity->code, $updatedEntity->code);
```
Comment (FrancisMurillo) [inline tests/CoreTest.php]: Repeat test please on another entity to prove `where` does not break the mapper

File: tests/CoreTest.php [hunk 2]
```diff
@@ -110,4 +110,44 @@ public function insertShouldWork()
...
+            foreach ($insertedEntities as $entity) {
+                $entities[] = $entity;
+            }
+        }
+
+        $mapper = self::$repo->entityMapper(CoreEntity::class);
+
+		$originalEntity = $entities[0];
+		$updatedEntityArray = CoreEntity::toRecord($entities[1]);
+		unset($updatedEntityArray["id"]);
```
[the remark this answers is not in the document]
Comment (James-A1) [reply tests/CoreTest.php]: Even if it's just the test?

File: src/Mapper.php [hunk 3]
```diff
@@ -177,4 +178,31 @@ public function insertRecord(array $entityRecord)
...
+     * @return object A new instance of the updated entity record.
+     */
+    public function update($entity)
+    {
+        $primaryField = $this->schema->getPrimaryField();
+        $fieldName = $primaryField->getFieldName();
+        $columnName = $primaryField->getColumnName();
+        $updatedFields = $this->entityClass::toRecord($entity);
+
+        $record = $this->query->cloneWithout([])->first();
```
[the remark this answers is not in the document]
Comment (James-A1) [reply src/Mapper.php]: 🙃 Yes, it shall be done

File: src/Mapper.php [hunk 4]
```diff
@@ -204,18 +204,28 @@ public function updateRecord(array $entityRecord)
...
             get_object_vars(
-                $this->query->cloneWithout([])->where($columnName, $entityRecord[$fieldName])->first()
+                $this->cloneQuery()->where($columnName, $entityRecord[$fieldName])->first()
             )
         );
 
 	}
+    
+    /**
+     * Clones the query property of the mapper
```
Comment (FrancisMurillo) [inline src/Mapper.php]: Add some extra comments why it cloned it. Like say to `avoid mutating the original query`

**Pull request merged**